### DISCUSSION TITLE
how do I add bookmark to the table of content?

### DISCUSSION THREAD

**Discussion by lcongdanh** [ahrm/sioyek#813] [Q&A] [answered]
some pdf does not have a table of content so I usually add bookmarks to make my own ToC. But that seems not the case in `Sioyek`. When I add a new bookmark, I have to find it with `gB`, so it's not in the ToC which can be opened with `t`. Is there a way to do it?

**Comment by ahrm** [comment, marked as answer]
No.

**Reply by lcongdanh** [reply]
I guess, I have something to contribute for this :)

**Comment by naveace** [comment]
i had basically the same question as this. @ahrm  do you know what would be necessary to add such a contribution? I am happy to try my hand at it in the coming months!

**Reply by ahrm** [reply]
It is not hard to implement, I just don't it is a common enough use-case to implement. You can search the bookmarks and table of contents separately using `gb` and `t` respectively.

**Reply by naveace** [reply]
I have had two use cases in my own work, although I can't speak for others of course:
1. If I want to read something on mobile since there is not any Sioyek for mobile (yet :) ), I will typically use in ibooks or some other pdf reader and that reader is unable to access the special Sioyek bookmarks so I lose all of them there
2. It is true that it is feasible to search seperately via `gb` and `t`, but often I find the auto-toc is *almost* right but not exactly right so then it turns into an awkward situation where most of the time using `t` is enough but sometimes I need to use `gb`. I think it would generally be a quality of life improvement to be able to just modify the TOC directly

Of course these area just nice-to-haves. The applications is great but I would love to help make it better :)
